src/plots/cartesian/axes.js

```diff
@@ -806,11 +806,11 @@ axes.autoTicks = function(ax, roughDTick) {
             roughDTick /= ONEAVGMONTH;
             ax.dtick = 'M' + roundDTick(roughDTick, 1, roundBase24);
         } else if(roughX2 > ONEDAY) {
-            ax.dtick = roundDTick(roughDTick, ONEDAY, roundDays);
-
             if(ax._hasDayOfWeekBreaks) {
-                if(ax.dtick === 2 * ONEDAY) ax.dtick = ONEDAY;
-                else if(ax.dtick === 3 * ONEDAY) ax.dtick = 7 * ONEDAY;
+                ax.dtick = roundDTick(roughDTick, ONEDAY, [1, 1.001, 7, 14]);
+                if(ax.dtick !== ONEDAY) ax.dtick = 7 * ONEDAY;
```

Comment:
Is this different from just saying `roundDTick(roughDTick, ONEDAY, [1, 7, 14])`? The underlying function is `Lib.roundUp` so I'd expect anything too big for 1 would already bump up to 7. Also I think this would convert 2-week ticks into 1-week?

Reply:
> Also I think this would convert 2-week ticks into 1-week?

Doog eye. Fixed in https://github.com/plotly/plotly.js/pull/4695/commits/79b6d95b73d727289b456616baea39db458b495c.

But we still need to fix something possibly in `Lib.dateTick0` to compute `dtick` on those `reversed` cases.

Reply:
again though, is this different from just `roundDTick(roughDTick, ONEDAY, [1, 7, 14])`?

Reply:
I think that is different. The issue was almost related to numbers close to 1 (e.g. 2) not those close to 7. Anyway it seems there is no more need for that hacy roundingSet as the logic is changed in the commits below.